data_steward/cdr_cleaner/cleaning_rules/deid/generate_site_mappings_and_ext_tables.py

```diff
@@ -0,0 +1,164 @@
+    src_id STRING OPTIONS(description="The masked id of the hpo site/RDR.")) AS (
+    WITH random_ids AS
+(
+  SELECT
+    CONCAT({{site_prefix}}, new_id) AS hpo_id,
+    ROW_NUMBER() OVER(ORDER BY GENERATE_UUID()) AS row_id
+  FROM
+    UNNEST(GENERATE_ARRAY(100, 999)) AS new_id
+)
+SELECT
+  r.hpo_id,
+  LOWER(m.hpo_id) as src_id
+FROM
+(
+  SELECT
+    m.*,
+    ROW_NUMBER() OVER(ORDER BY GENERATE_UUID()) AS row_id
+  FROM `{{project_id}}.{{lookup_tabels_dataset}}.{{hpo_site_id_mappings_table}}` AS m
+  WHERE m.HPO_ID IS NOT NULL AND m.HPO_ID != ''
+) AS m
+JOIN random_ids AS r
+USING (row_id)
+union all
+select {{ppi_pm}} as hpo_id, {{rdr}} as src_id
+)
+""")
+
+
+class GenerateSiteMappingsAndExtTables(BaseCleaningRule):
+    """
+    Generates non-deterministic site_ids for hpo sites.
+    """
+
+    def __init__(self, project_id, dataset_id, sandbox_dataset_id,
+                 mapping_dataset_id):
+        """
+        Initialize the class with proper info.
+
+        Set the issue numbers, description and affected datasets.  As other
+        tickets may affect this SQL, append them to the list of Jira Issues.
+        DO NOT REMOVE ORIGINAL JIRA ISSUE NUMBERS!
+        """
+        desc = f'Change PIDs to RIDs in specified tables'
+
+        self.mapping_dataset_id = mapping_dataset_id
+
+        super().__init__(issue_numbers=ISSUE_NUMBERS,
+                         description=desc,
+                         affected_datasets=[cdr_consts.CONTROLLED_TIER_DEID],
+                         project_id=project_id,
+                         dataset_id=dataset_id,
+                         sandbox_dataset_id=sandbox_dataset_id,
+                         affected_tables=[])
+
+    def get_query_specs(self):
+        """
+        Return a list of dictionary query specifications.
+
+        :return:  A list of dictionaries.  Each dictionary contains a
+            single query and a specification for how to execute that query.
+            The specifications are optional but the query is required.
+        """
+        query_list = []
+        query = dict()
+        query[cdr_consts.QUERY] = SITE_MAPPINGS_QUERY.render(
+            project_id=self.project_id,
+            dataset_id=self.sandbox_dataset_id,
+            table_id=SITE_TABLE_ID,
+            site_prefix=EHR_SITE_PREFIX,
+            lookup_tabels_dataset=LOOKUP_TABLES_DATASET_ID,
+            hpo_site_id_mappings_table=HPO_SITE_ID_MAPPINGS_TABLE_ID,
+            ppi_pm=PPI_PM,
+            rdr=RDR)
+        query_list.append(query)
+
+        # gather queries to generate ext tables
+        query_list.append(
+            get_generate_ext_table_queries(self.project_id,
+                                           self.sandbox_dataset_id,
+                                           self.sandbox_dataset_id,
+                                           self.mapping_dataset_id))
```

Comment:
`get_generate_ext_table_queries` returns a list, use extend instead of append
```suggestion
        query_list.extend(
            get_generate_ext_table_queries(self.project_id,
                                           self.sandbox_dataset_id,
                                           self.sandbox_dataset_id,
                                           self.mapping_dataset_id))
```